src/m_menu.c

```diff
@@ -3795,47 +3800,57 @@ static void M_UpdateFOV(void)
 
 static void M_ResetScreen(void)
 {
-  need_reset = true;
+  resetneeded = true;
+}
+
+static void M_UpdateSfxVolume(void)
+{
+  S_SetSfxVolume(snd_SfxVolume);
+}
+
+static void M_UpdateMusicVolume(void)
+{
+  S_SetMusicVolume(snd_MusicVolume);
 }
 
 setup_menu_t gen_settings1[] = { // General Settings screen1
 
   {"Video"       ,S_SKIP|S_TITLE, m_null, M_X, M_Y},
 
-  {"Resolution Mode", S_CHOICE, m_null, M_X, M_Y+ gen1_hires*M_SPC,
-   {"resolution_mode"}, 0, M_ResetScreen, resolution_mode_strings},
+  {"Resolution Scale", S_THERMO|S_THRM_SIZE12|S_ACTION, m_null, M_X_THRM12, M_SPC,
+   {"resolution_scale"}, 0, M_ResetVideoHeight, str_resolution_scale},
+
+  {"Dynamic Resolution", S_YESNO, m_null, M_X, M_THRM_SPC,
+   {"dynamic_resolution"}, 0, M_ResetVideoHeight},
 
-  {"Widescreen Rendering", S_YESNO, m_null, M_X, M_Y+ gen1_widescreen*M_SPC,
+  {"Widescreen Rendering", S_YESNO, m_null, M_X, M_SPC,
    {"widescreen"}, 0, M_ResetScreen},
 
-  {"FOV", S_NUM, m_null, M_X, M_Y + gen1_fov*M_SPC,
-   {"fov"}, 0, M_UpdateFOV},
+  {"FOV", S_THERMO, m_null, M_X_THRM8, M_SPC, {"fov"}, 0, M_UpdateFOV},
 
-  {"", S_SKIP, m_null, M_X, M_Y + gen1_gap1*M_SPC},
+  {"", S_SKIP, m_null, M_X, M_THRM_SPC},
 
-  {"Fullscreen Mode", S_YESNO, m_null, M_X, M_Y+ gen1_fullscreen*M_SPC,
+  {"Fullscreen Mode", S_YESNO, m_null, M_X, M_SPC,
```

Comment:
Some more thoughts discussed in the past but worth considering again to declutter the menu:

1. This may be a good opportunity to combine the "Fullscreen" and "Exclusive Fullscreen" settings into one, named "Display Mode":

```
Display Mode:

Menu:                 woof.cfg:
Windowed              fullscreen 0 (was fullscreen 0)
Borderless (default)  fullscreen 1 (was fullscreen 1 + exclusive_fullscreen 0)
Fullscreen            fullscreen 2 (was fullscreen 1 + exclusive_fullscreen 1)
```

2. Same with "Uncapped Framerate" and "Framerate Limit":

```
Framerate Limit:

Menu:        woof.cfg:
0 (default)  fpslimit 0      (was uncapped 1 + fpslimit 0)
35           fpslimit 35     (was uncapped 1 + fpslimit 35, or uncapped 0)
36-500       fpslimit 36-500 (was uncapped 1 + fpslimit 36-500)
```

Reply:
> I would prefer to keep Excusive Fullscreen separate. It seems to depend on the video driver/Windows version, but on Windows 10 I still have some glitches sometimes.

It's still separate but maybe my table was confusing. Here's a better one:

| Fullscreen | Exclusive Fullscreen | --> | Display Mode                          | `fullscreen` |
|------------|----------------------|-----|---------------------------------------|--------------|
| No         | Ignored              | --> | Windowed                              | `0`          |
| Yes        | No                   | --> | Fullscreen (or Borderless Fullscreen) | `1`          |
| Yes        | Yes                  | --> | Exclusive Fullscreen                  | `2`          |

```
Old Setting:
Fullscreen: No, Exclusive Fullscreen: (Ignored)
```

> Should we make a thermo bar for framerate?

A thermo bar is okay, but only if the behavior is modified so that it increases or decreases faster the longer the keyboard/gamepad input is held down. Otherwise it'll be very slow moving through the range (this is already a problem with some thermos like the gamepad sensitivities). A number entry for framerate is fine if that seems like too much work.

Reply:
> It's still separate but maybe my table was confusing. Here's a better one:

Thanks for the table, but we need to use a multi-choice element (`S_CHOICE`) for this, right? At the moment it's too easy to toggle with the mouse, so user might accidentally switch Exclusive Fullscreen on and have a bad experience. I think we should make this switch more intimidating 😄 

> A thermo bar is okay, but only if the behavior is modified so that it increases or decreases faster the longer the keyboard/gamepad input is held down.

For the gamepad, it's easy since I've already implemented repetition in the menu responder. For the keyboard, I rely on SDL. Alternatively, we can gradually increase the steps. But perhaps entering a number is more convenient here.